src/CodeGenerator/AbstractInjector.php

```diff
@@ -49,14 +49,21 @@ public function __construct(InjectorInterface $injector, ContainerInterface $con
      */
     abstract protected function loadFactoryList() : void;
 
+    private function instantiateFactory(string $factoryClass) : FactoryInterface
+    {
+        return new $factoryClass();
```

Comment:
Probably requires an `assert()` call

Reply:
I thought about assert() as well, but it will not enforce type correctness. This way the call will always fail on incorrect type. Maybe I can refactor it differently when working on the typechange issue of the factories member.

Reply:
:+1: I thought the static analysis would clog up here

Reply:
It did, but assert is too weak here imho.